src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -321,28 +321,21 @@ private void Initialize(XmlReader reader, string fileName, bool ignoreErrors)
             catch (Exception exception)
             {
                 InitializeSucceeded = false;
-                if (exception.MustBeRethrown())
+                if (exception.MustBeRethrownImmediately())
                 {
                     throw;
                 }
 
-                NLogConfigurationException ConfigException = new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception);
+                InternalLogger.Error(new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception), "Error in Parsing Configuration File.");
 
                 if (!ignoreErrors)
                 {
-                    if (LogManager.ThrowExceptions)
-                    {
-                        throw ConfigException;
-                    }
-                    else
+                    if (exception.MustBeRethrown())
```

Comment:
Because `exception` is an inner-exception of `configurationException` it will be written to the log with the `InternalLogger.Error()` call above, but not be marked as logged. Therefore `exception.MustBeRethrown()` will log it for the second time. Perhaps `MarkAsLoggedToInternalLogger()` must be changed to recursively mark all inner exceptions as well.